### PR TITLE
Delete roles

### PR DESCRIPTION
**Pull request by jsteiner**

### CODE REVIEW

File: lib/cog_api/http/base.ex [hunk 1]
```diff
@@ -27,13 +27,7 @@ defmodule CogApi.HTTP.Base do
...
   def delete(%Endpoint{}=endpoint, resource) do
     rescue_econnrefused(fn ->
-      response = HTTPotion.delete(make_url(endpoint, resource), headers: make_headers(endpoint))
-      case response_type(response) do
-        :ok ->
-          :ok
-        :error ->
-          {:error, Poison.decode!(response.body)}
-      end
+      HTTPotion.delete(make_url(endpoint, resource), headers: make_headers(endpoint))
```
Comment (drapergeek) [inline lib/cog_api/http/base.ex]: Why did this change? Are we not handling the failed cases?

File: lib/cog_api/http/roles.ex [hunk 2]
```diff
@@ -21,4 +21,9 @@ defmodule CogApi.HTTP.Roles do
     patch(endpoint, "roles/#{role_id}", %{"role" => params})
     |> format_response("role", %Role{})
   end
+
+  def role_delete(%Endpoint{}=endpoint, role_id) do
+    delete(endpoint, "roles/#{role_id}")
```
Comment (drapergeek) [inline lib/cog_api/http/roles.ex]: One line? Not picky, just an option.

File: test/unit/cog_api/fake/roles_test.exs [hunk 3]
```diff
@@ -69,4 +69,21 @@ defmodule CogApi.Fake.RolesTest do
...
+      {:ok, role} = Roles.role_create(fake_endpoint, %{name: "new role"})
+
+      Roles.role_delete(fake_endpoint, role.id)
+
+      {:ok, roles} = Roles.role_index(fake_endpoint)
+
+      refute Enum.member?(roles, role)
+    end
+
+    it "returns :ok" do
```
Comment (drapergeek) [inline test/unit/cog_api/fake/roles_test.exs]: Is this necessary? Could we just put this in with the other test?

Comment (drapergeek) [inline test/unit/cog_api/fake/roles_test.exs]: Nevermind, I see the other test.

**Comment by drapergeek** [comment]
LGTM

File: lib/cog_api/http/base.ex (hunk 1 above)
Comment (jsteiner) [inline lib/cog_api/http/base.ex]: That's being handled by `format_response` by the calling function, which brings it up to spec with `get`, `post` and `patch`

**Pull request merged**